volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -498,6 +593,66 @@ def get_mnt_parent(self):
     def get_mnt_mountpoint(self):
         return self.mnt_mountpoint
 
+    def get_flags_access(self) -> str:
```

Comment:
These seems useful, so I'm willing to accept them as a change to the core files as long as @atcuno is happy with them, since they'll be a change to the API.  This, as an additive change to the API, should also include a bump to the minor framework version in the `constants/__init__.py` file.  This is so that plugins that need it can say they need at least this version to operate.  Your plugin making use of these should also require the bumped version of the framework.  No other plugins should need to be changed though...

Reply:
We can move the constants to the constants file. I thought maybe it could look more organised to have them encapsulated in their class given them even more context but not too fussy with it.

Reply:
Yep, that's fine, what I was saying about the constants file is that we need to bump the framework minor version (making it `(2, 1, 0)` since these are core changes that people might want to rely on in the future.  They need to be able to say their plugin will work with 2.1.0 but not 2.0.0, so just include a bump at https://github.com/volatilityfoundation/volatility3/blob/develop/volatility3/framework/constants/__init__.py#L42 (`VERSION_MINOR`) please...

Reply:
oh ok. Does it mean I should also bump the MountInfo `_required_framework_version` version? After changing the VERSION_MINOR = 1 it still works even using _required_framework_version = (2, 0, 0)

Reply:
Oh ok, I checked it. As per the [code](https://github.com/volatilityfoundation/volatility3/blob/3c11099e6621027777ac5a9eeb3f8fbdc5b39b4b/volatility3/framework/__init__.py#L50), the major only has to be the same. The minor module version requirement could be same or lower and the patch level is not checked at all.
hm, I think it would be good idea to use something like a standard python module [packaging.version](https://packaging.pypa.io/en/latest/version.html) for comparing versions instead of doing it manually.

Also, note that the exception message is not accurate. In the hypothetical case that the framework has a lower version than the module requirement, the message would be, for instance:
'Framework interface version 2 is an older revision than the required version 2.2'
instead of:
'Framework interface version 2.1 is an older revision than the required version 2.2'

```diff
--- /tmp/qRIePE___init__.py
+++ /home/gmoreira/vol3-latest/volatility3/framework/__init__.py
@@ -51,7 +51,7 @@
             if args[1] > interface_version()[1]:
                 raise RuntimeError(
                     "Framework interface version {} is an older revision than the required version {}".format(
-                        ".".join([str(x) for x in interface_version()[0:1]]), ".".join([str(x) for x in args[0:2]])))
+                        ".".join([str(x) for x in interface_version()[0:2]]), ".".join([str(x) for x in args[0:2]])))
 
 
 class NonInheritable(object):
```

Reply:
> Does it mean I should also bump the MountInfo `_required_framework_version` version? 

In the same commit I also bumped the module required framework version. Let me know if that's not necessary.

Reply:
Thanks, if `mountinfo` needs the features that you're adding (that wouldn't be present in `(2, 0, 0)`) then yes, you'd need to bump it.

As to using packaging.version, that would require a large external dependency for something we've already got code for.  It also has a lot more options than we need.  We also use semantic versioning with is mostly compatible with PEP 440, but again, why drag in a big dependency for something that's mostly compatible with what we already use.  I'm not against using a package for versioning, but I haven't heard a compelling reason to do so yet...